common-messaging/src/main/scala/org/genivi/sota/messaging/MessageBus.scala

```diff
@@ -76,6 +77,32 @@ object MessageBus {
     }
   }
 
+  def subscribeCommittable[T <: BusMessage](config: Config)
+                                           (implicit messageLike: MessageLike[T], system: ActorSystem)
+  : Source[CommittableMessageLike[T], NotUsed] = {
+    config.getString("messaging.mode").toLowerCase().trim match {
+      case "nats" =>
+        log.info("Starting messaging mode: NATS")
+        log.info(s"Using subject name: ${messageLike.streamName}")
+        NatsClient.source(system, config, messageLike.streamName)(messageLike.decoder) match {
```

Comment:
It is a bit weird that if I do the `subscribeCommittable` I could get a non-committable `Source` depending on `config.getString("messaging.mode")`. I don't really have another solution at the moment.